src/main/java/io/jenkins/plugins/coverage/targets/CoverageResult.java

```diff
@@ -502,6 +503,17 @@ public CoverageResult getPreviousResult() {
 
     public Object getDynamic(String token, StaplerRequest req, StaplerResponse rsp) throws IOException {
         token = token.toLowerCase();
+        String restPath = req.getRestOfPath();
+
+        // prevent name conflict
+        if (restPath.startsWith("/api/")) {
+            if (token.equals("trend")) {
+                return new RestResultWrapper(new CoverageTrendTree(getName(), getCoverageTrends(), getChildrenReal()));
+            } else if (token.equals("result")) {
+                return new RestResultWrapper(this);
```

Comment:
why do you need a wrapper here? it seems yo introduce recursive API

Reply:
@oleg-nenashev If I return `new Api(this)` directly, we need to retrive data by URL like
 `.../result/json` instead of `.../result/api/json`. 
And I do not think it will cause a recursive call, one token map one object...